geode-wan/src/main/java/org/apache/geode/internal/cache/wan/GatewaySenderEventRemoteDispatcher.java

```diff
@@ -348,11 +349,56 @@ public void destroyConnection() {
     }
   }
 
+  Connection retryInitializeConnection(Connection con) {
+    final boolean isDebugEnabled = logger.isDebugEnabled();
+    ServerLocation server = this.sender.getServerLocation();
+    String connectedServerId = con.getEndpoint().getMemberId().getUniqueId();
+    String expectedServerId = this.processor.getExpectedReceiverUniqueId();
+    boolean connectedToExpectedReceiver = connectedServerId.equals(expectedServerId);
+    if (expectedServerId.equals("")) {
+      if (isDebugEnabled) {
+        logger.debug("First dispatcher connected to endpoint " + connectedServerId);
+      }
+      this.processor.setExpectedReceiverUniqueId(connectedServerId);
+      connectedToExpectedReceiver = true;
+    }
+    int attempt = 0;
+    final int maxAttempts = 5;
```

Comment:
When this method is invoked, do we expect that the first call to `acquireConnection()` should result in a connection to the desired server, or is there an element of randomness in which server we end up getting connected to, hence the retries? If it's the latter, this method could prove unusable for situations in which the chance of getting the correct server at random is low.

Reply:
If several receivers are sharing the same ip and port, the connection will be assigned randomly by an external proxy or load balancer. Its true that using this fix value is not the best option... If for example we have 5 receivers, we could spend all the connection retries without reaching the desired receiver.
Maybe I could increment the value of `maxAttempt` when a connection to a wrong receiver is obtained and its the first time we reach that wrong receiver. This will cause that `maxAttempt` value is dependent on the number of receivers.

Reply:
I feel a little uneasy about the idea of something that could fail at random just due to bad luck. There's a non-zero (albeit small) chance that we would never get the "right" receiver using this approach, and the number of retries necessary to ensure we have some minimum probability of getting the right receiver increases rapidly with each receiver that's sharing the same ip and port.

Reply:
With the approach I was suggesting, we will have `x * y ` attempts to connect to the desired receiver, being `x=5` (or any other number) and `y=<number of receivers under same ip and port>`.

I think this is a high enough number to guarantee we reach the receiver we want. As you said, we are not setting the probability of not reaching the proper server to zero, but I think we are reducing it to a reasonable value.

I will add a commit with the change.